src/signTxUtils.c

```diff
@@ -3,10 +3,39 @@
 #include "uiHelpers.h"
 #include "utils.h"
 #include "signTxUtils.h"
+#include "securityPolicy.h"
+#include "state.h"
+
+static single_account_data_t* singleAccountData = &(instructionState.signTxContext.commonTxData.singleAccountData);
 
 void respondSuccessEmptyMsg()
 {
 	TRACE();
 	io_send_buf(SUCCESS, NULL, 0);
 	ui_displayBusy(); // needs to happen after I/O
 }
+
+bool violatesAndStoreSingleAccount(const bip44_path_t* path)
+{
+	if (!bip44_hasOrdinaryWalletKeyPrefix(path) || !bip44_containsAccount(path)) {
+		TRACE("Invalid path in single account check");
+		ASSERT(false);
+	}
+	const bool byron = bip44_hasByronPrefix(path);
+	const uint32_t account = bip44_getAccount(path);
+	if (singleAccountData->pathSet) {
+		const uint32_t storedAccount = singleAccountData->accountNumber;
+		if (account != storedAccount) {
+			return true;
+		}
+		if ((singleAccountData->byron ^ byron)
+		    && storedAccount != 0) {
```

Comment:
```suggestion
		bool combinesByronAndShelley = singleAccountData->byron ^ byron;
		bool combinationAllowed = (storedAccount == 0);
		if (combinesByronAndShelley && !combinationAllowed) {
```

Reply:
Wouldn't it more readable to use `singleAccountData->byron != byron` instead of `singleAccountData->byron ^ byron`? If I am not mistaken the result should be the same.